app/controllers/admin/product_import_controller.rb

```diff
@@ -36,7 +39,22 @@ def validate_data
     def save_data
       return unless process_data('save')
 
-      render json: @importer.save_results
+      json = {
+        results: {
+          products_created: @importer.products_created_count,
+          products_updated: @importer.products_updated_count,
+          products_reset: @importer.products_reset_count,
+        },
+        updated_ids: @importer.updated_ids,
+        errors: @importer.errors.full_messages
+      }
```

Comment:
Nice, it's probably better to have the json format defined here in the controller than in the class 👍